cloud/pkg/cloudstream/streamserver.go

```diff
@@ -109,23 +109,12 @@ func (s *StreamServer) getContainerLogs(r *restful.Request, w *restful.Response)
 		}
 	}()
 
-	// extract pod namespace and pod name from request
-	meta := strings.Split(r.Request.URL.Path, "/")
-	namespaceName := meta[2]
-	podName := meta[3]
-
-	kubeClient := client.GetKubeClient()
-	if kubeClient == nil {
-		klog.Exitf("cannot get kube client\n")
-		return
-	}
-
-	pod, err := kubeClient.CoreV1().Pods(namespaceName).Get(context.Background(), podName, v1.GetOptions{})
+	sessionKey, err := s.getSessionKey(r.Request.URL.Path)
 	if err != nil {
-		klog.Errorf("get pod %s/%s failed: %v\n", namespaceName, podName, err)
+		err = fmt.Errorf("can not get session key: %v", err)
```

Comment:
```suggestion
		klog.Errorf("Failed to get session key, err: %v", err)
```
We need to print the err log?

Reply:
The err log seems to be printed at defer statement?

https://github.com/kubeedge/kubeedge/blob/0ec91cf419ddfac72da8d840ee1c03018ffc76b5/cloud/pkg/cloudstream/streamserver.go#L105-L110

Reply:
The error will not be print in func `getSessionKey`, it should be printed here.
And if we don't print here, what's the usage of this line `err = fmt.Errorf("can not get session key: %v", err)`? and the next line is return :)

Reply:
The error will be printed in the `defer` clause:

https://github.com/kubeedge/kubeedge/blob/0ec91cf419ddfac72da8d840ee1c03018ffc76b5/cloud/pkg/cloudstream/streamserver.go#L103-L116

I did an experiment locally and confirm the behavior that the err would be printed at https://github.com/kubeedge/kubeedge/blob/0ec91cf419ddfac72da8d840ee1c03018ffc76b5/cloud/pkg/cloudstream/streamserver.go#L108

Similarly, other existing error reporting logic has the same pattern. For example: 

https://github.com/kubeedge/kubeedge/blob/0ec91cf419ddfac72da8d840ee1c03018ffc76b5/cloud/pkg/cloudstream/streamserver.go#L128

I don't know why the error printing logic is designed like this previously. But it works.

Reply:
Ohh, sorry, I missed the log in defer which you also mentioned before...  And can you fix the error meessage, it need starts with a uppercase.

Reply:
error starts with a lowercase, log info starts with a uppercase
So no need to fix case :)

Reply:
Yes, per [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments#error-strings), error starts with a lowercase:

> Error strings should not be capitalized (unless beginning with proper nouns or acronyms) or end with punctuation, since they are usually printed following other context. That is, use fmt.Errorf("something bad") not fmt.Errorf("Something bad"), so that log.Printf("Reading %s: %v", filename, err) formats without a spurious capital letter mid-message. This does not apply to logging, which is implicitly line-oriented and not combined inside other messages.

Reply:
This err will be printed directly in L108

Reply:
Or we can modify L108 as: klog.Errorf("Failed to get container logs, err: %v", err)

Reply:
done. `modify L108 as: klog.Errorf("Failed to get container logs, err: %v", err)` should be the right way :)